src/App.tsx

```diff
@@ -1,156 +1,213 @@
+                  <button
+                    type="button"
+                    className="todo__remove"
+                    onClick={() => handleDeleteTodo(todo.id)}
+                  >
+                    ×
+
+                  </button>
+
+                  {/* overlay will cover the todo while it is being updated */}
+                  <div className="modal overlay">
+                    <div className="modal-background has-background-white-ter" />
+                    <div className="loader" />
+                  </div>
+                </div>
+              );
+            })}
+          </section>
+
+          {/* Hide the footer if there are no todos */}
+          {todos.length !== 0 && (
+            <footer className="todoapp__footer">
+              <span className="todo-count">
+                {`${visibleTodos.filter(todo => !todo.completed).length} items left`}
+              </span>
+
+              {/* Active filter should have a 'selected' class */}
+              <nav className="filter">
+                <a
+                  href="#/"
+                  className={classNames('filter__link', {
+                    selected: filter === 'all',
+                  })}
+                  onClick={() => setFilter('all')}
+                >
+                  All
+                </a>
+
+                <a
+                  href="#/active"
+                  className={classNames('filter__link', {
+                    selected: filter === 'active',
+                  })}
+                  onClick={() => setFilter('active')}
+                >
+                  Active
+                </a>
+
+                <a
+                  href="#/completed"
+                  className={classNames('filter__link', {
+                    selected: filter === 'completed',
+                  })}
+                  onClick={() => setFilter('completed')}
+                >
+                  Completed
+                </a>
+              </nav>
+
+              {/* don't show this button if there are no completed todos */}
+              <button type="button" className="todoapp__clear-completed">
+                Clear completed
+              </button>
+            </footer>
+          )}
+        </div>
+      )}
 
       {/* Notification is shown in case of any error */}
       {/* Add the 'hidden' class to hide the message smoothly */}
-      <div className="notification is-danger is-light has-text-weight-normal">
-        <button type="button" className="delete" />
-
-        {/* show only one message at a time */}
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
-        <br />
-        Unable to update a todo
-      </div>
+      {hasError && (
+        <div className="notification is-danger is-light has-text-weight-normal">
+          <button
+            type="button"
+            className="delete"
+            onClick={() => setHasError(false)}
+          />
+
+          Unable to add a todo
```

Comment:
Consider showing specific error messages for different error cases such as 'Unable to add a todo', 'Unable to delete a todo', and 'Unable to update a todo'. You may want to use a state variable to store the error message and display it here.